Add interactive mode

fliepeltje opened the pull request:
Adds an interactive flag that allows you to keep the conversation open. 

![image](https://user-images.githubusercontent.com/49225672/226416263-29767594-f4ae-45ee-8bdb-943ab1c529d7.png)

npiv commented:
Hey @fliepeltje thx for the contribution. I like the idea, and the implementation. I played with it a bit on your branch:

Correct me if I'm wrong but currently there is no way to exit without `ctrl-c` which produces a nasty strack trace. It would be cool if `exit` or `quit` took you out. Or additionally if ctrl-c didn't spew to the console could also be an option

### nice to haves (not blocking)
- does rich offer any way to enter multiple lines? Part of the benefit of an interactive mode would be to enter long texts, and maybe then ctrl-enter would send it off. Would be curious for your ideas on that?
-  just typing `-i` on its own could ask you for a prompt and you could start without a query, or you could just load your promptconfig and interactively enter the query

fliepeltje commented:
I think a multiline input would require playing around with curses, which is unfortunately not something I am familiar with. I've added an exception for keyboard interrupts that gracefully exits the cli if the interrupt happens during the prompt. Additionally from a usability point of view I added the option to quit by typing `quit` as per your suggestion; I think that is a nice interface. 

Additionally if you start the cli with just the flag, it will prompt you for a query (`chatblade --interactive`)

npiv commented:
Awesome thanks!

Are you dutch by any chance? I'm Flemish myself :) Just going of your name . I've merged the change let me know if you would be interested in becoming a co-maintainer. The project got a lot of attention today, and I have a few ideas for the long term, would be cool to have someone to bounce ideas with and grow functionality.

We could have a chat this week if interested.
